source/include/ice_api.h

```diff
@@ -0,0 +1,105 @@
+#ifndef ICE_API_H
+#define ICE_API_H
+
+#include "ice_data_types.h"
+
+IceResult_t Ice_CreateIceAgent( IceAgent_t * pIceAgent,
+                                char * pLocalUsername,
+                                char * pLocalPassword,
+                                char * pRemoteUsername,
+                                char * pRemotePassword,
+                                char * pCombinedUsername,
+                                TransactionIdStore_t * pBuffer,
+                                Ice_ComputeRandom computeRandomFunction,
+                                Ice_ComputeCrc32 computeCrc32Function,
+                                Ice_ComputeHMAC computeHMACFunction );
+
+IceResult_t Ice_AddHostCandidate( const IceIPAddress_t ipAddr,
+                                  IceAgent_t * pIceAgent,
+                                  IceCandidate_t ** ppCandidate );
+
+IceResult_t Ice_AddSrflxCandidate( const IceIPAddress_t ipAddr,
+                                   IceAgent_t * pIceAgent,
+                                   IceCandidate_t ** ppCandidate,
+                                   uint8_t * pTransactionIdBuffer,
+                                   uint8_t ** ppSendStunMessageBuffer,
+                                   uint32_t * pSendStunMessageBufferLength );
+
+IceResult_t Ice_AddRemoteCandidate( IceAgent_t * pIceAgent,
+                                    IceCandidateType_t iceCandidateType,
+                                    IceCandidate_t ** ppCandidate,
+                                    const IceIPAddress_t ipAddr,
+                                    IceSocketProtocol_t remoteProtocol,
+                                    const uint32_t priority );
+
+IceResult_t Ice_CreateCandidatePair( IceAgent_t * pIceAgent,
+                                     IceCandidate_t * pLocalCandidate,
+                                     IceCandidate_t * pRemoteCandidate );
+
+IceResult_t Ice_UpdateSrflxCandidateAddress( IceAgent_t * pIceAgent,
```

Comment:
Same here. Can we move this to other header file to avoid user including this unexpectly?